airbyte-integrations/connectors/source-avni/source_avni/source.py

```diff
@@ -0,0 +1,180 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+
+import boto3
+import requests
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import IncrementalMixin, Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+
+
+class AvniStream(HttpStream, ABC):
+
+    url_base = "https://app.avniproject.org/api/"
+    primary_key = "ID"
+
+    def __init__(self, lastModifiedDateTime: str, auth_token: str, **kwargs):
+        super().__init__(**kwargs)
+        self.cursor_value = None
+        self.current_page = 0
+        self.lastModifiedDateTime = lastModifiedDateTime
+        self.last_record = None
+        self.auth_token = auth_token
```

Comment:
Usually connector set this to class level because they're not going to be instantiate ever.
```suggestion
class AvniStream(HttpStream, ABC):

    url_base = "https://app.avniproject.org/api/"
    primary_key = "ID"
    self.cursor_value = None
    self.current_page = 0

    def __init__(self, start_date: str, auth_token: str, **kwargs):
        super().__init__(**kwargs)
        self.start_date = start_date
        self.auth_token = auth_token
```

Reply:
modified as you suggested